backend/src/services/data_fetcher.py

```
@@ -14,23 +14,71 @@ class DataFetcherService:
+            else:
+                url = f"{DEXSCREENER_BASE_URL}/token-boosts/top/v1"
+
+            response = await self._request_with_retry("GET", url)
+            if not response:
+                return []
+
             data = response.json()
             return self._process_dexscreener_data(data)
-        except httpx.RequestError as e:
-            logger.error(f"HTTP error fetching from Dexscreener: {e}")
         except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)